src/test/java/AnimalTest.java

```diff
@@ -0,0 +1,102 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static java.lang.System.out;
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+
+
+public class AnimalTest {
+
+    @Test
+    void testOrientation() {
+        Animal cat = new Animal();                      //orientation and position are hardcoded in the class Animal
+        //   System.out.println(kot.toString());
+        MoveDirection[] directions = new MoveDirection[]{
+                MoveDirection.LEFT,
+                MoveDirection.RIGHT,
+                MoveDirection.RIGHT,
+                MoveDirection.RIGHT};
+        MapDirection[] directionsWorldSides = new MapDirection[]{
+                MapDirection.WEST,
+                MapDirection.NORTH,
+                MapDirection.EAST,
+                MapDirection.SOUTH};
+        for (int i = 0; i < directions.length; i++) {
+            cat.move(directions[i]);
+            assertEquals(directionsWorldSides[i], cat.getOrientation());
+        }
+    }
+
+    // testOrientation and testPosition could be merged in one, but I was wondering what is clearer, and more practical
+    // when some part of the function changes, and decided to left separated
+    @Test
+    void testPosition(){
+            Animal cat = new Animal();                      //orientation and position are hardcoded in the class Animal
+            //   System.out.println(kot.toString());
+            MoveDirection[] directions = new MoveDirection[]{
+                    MoveDirection.FORWARD,
+                    MoveDirection.BACKWARD,
+                    MoveDirection.RIGHT,
+                    MoveDirection.FORWARD,
+                    MoveDirection.BACKWARD,
+                    MoveDirection.RIGHT,
+                    MoveDirection.FORWARD,
+                    MoveDirection.BACKWARD,
+                    MoveDirection.RIGHT,
+                    MoveDirection.FORWARD,
+                    MoveDirection.BACKWARD,
+                    MoveDirection.RIGHT
+                    };
+            for (int i = 0; i < directions.length; i++) {
+                cat.move(directions[i]);}
+                assertEquals(true, cat.isAt(new Vector2d(2,2)));
```

Comment:
ok, przy czym w tego typu teście sprawdzamy rezultat końcowy całej sekwencji - jeśli po drodze program popełniłby kilka sprzecznych ze sobą błędów i wyszłoby na końcu na plus to test dalej by przechodził.